Fix: clarify managed v. deployed certs (#291)
nginx/documentation: content/nginx-one/how-to/certificates/manage-certificates.md

The Export option supports exports of basic certification file information to a CSV file. It does _not_ include the content of the public certificate or the private key.

## Deployed certificates

> A deployed certificate is associated with a specific Instance or Config Sync Gruop. That certificate may be Managed or Unmanaged.

Reviewer: ```suggestion
A deployed certificate is associated with a specific Instance or Config Sync Group. That certificate may be Managed or Unmanaged.
```

Author: Updated, https://github.com/nginx/documentation/pull/291/files#diff-2445ce44d5814d8ceb795a09fa3672417a9d86c176326cf9ecbc3036b8fc5702R136

If that certificate is already managed as part of a Config Sync Group, the changes you make affect all instances in that group.

## Remove a managed certificate

> You can remove a managed certificate from an independent instance or from a Config Sync Group. This will remove the certificate's association with the Instance or group, but it does not delete the certificate files from the Instance(s).

Reviewer 2: ```suggestion
From the configuration editor where you normally modify NGINX configuration files of an independent instance or Config Sync Group, you can click on the "delete" icon of a managed certificate object that was previously deployed to the instance or Config Sync Group. You should be able to see the file paths where it was deployed to. Deleting the managed certificate from the NGINX configuration editor will remove the certificate files from those file paths. If the certificate object is a certificate-key pair, and the private key was deployed, you could optionally choose to delete the deployed key from the independent instance or Config Sync Group, by clicking on the "delete" icon next to the private key file.
```

Here is a rough summary of what users could do. Feel free to change the wording and improve this paragraph further! : )

Author: ```suggestion
You can delete managed certificates in the following ways:

 -  Navigate to [View and edit NGINX configurations]({{< relref "/nginx-one/how-to/nginx-configs/view-edit-nginx-configurations/" >}})
    - You can then delete the certificate from the Instance of your choice.
- Navigate to [Manage Config Sync Groups]({{< relref "/nginx-one/how-to/config-sync-groups/manage-config-sync-groups" >}})
    - You can then delete the certificate from the Config Sync Group of your choice.
- Review the list of existing certificates
    - From the **Actions** menu, you can then delete that certificate
```

Reviewer 2: > - Review the list of existing certificates
>     - From the **Actions** menu, you can then delete that certificate
This would be incorrect. The contexts here are that users can remove a deployed certificate file from an instance or a CSG. They should only be able to do that through the config editor (where users edit NGINX configuration for an instance or a CSG). It would also be helpful to clarify that when users click on the delete icon from the config editor, they would only delete the deployed certificate or key file under **specific file paths**.

https://frontdoor-test-docs.nginx.com/previews/docs/291/nginx-one/how-to/nginx-configs/manage-config-sync-groups/
In addition, I wasn't able to find any instructions on how users could delete the certificate from a Config Sync Group in this page, so this could be confusing to the reader.

I'm thinking that deleting a certificate file is a special case for deleting an aux file. Since we have a page that documents how to add a file, should we also add a new page that documents how to remove a file? That might be easier in comparison to trying to fit all the details into a small section in the cert management doc.
https://frontdoor-test-docs.nginx.com/previews/docs/291/nginx-one/how-to/nginx-configs/add-file/
https://frontdoor-test-docs.nginx.com/previews/docs/291/nginx-one/how-to/config-sync-groups/add-file-csg/

Author: I'm realizing that our coverage of "Edit Configuration" for instances and CSGs are haphazard. I'm going to address @sylwang 's comments, and later, address https://github.com/nginx/documentation/issues/427 to "unify" the messaging.

Author: @sylwang see: https://github.com/nginx/documentation/pull/291/files#diff-2445ce44d5814d8ceb795a09fa3672417a9d86c176326cf9ecbc3036b8fc5702R170-R178

Reviewer 2: Thank you for opening a new PR to address this. The new changes look good!

> You can remove a managed certificate from an independent instance or from a Config Sync Group. This will remove the certificate's association with the Instance or group, but it does not delete the certificate files from the Instance(s).

Author: ```suggestion
You can remove a managed certificate from an independent instance or from a Config Sync Group. This will remove the certificate's association with the Instance or group. It also deletes the certificate files from the data plane Instance(s).
```

> Every Instance with a certificate includes paths to certificates in their configuration files. If you remove one certificate, that change is limited to that one Instance.

Author: We'll leave it out :) 

TLDR: it's important to let the UI do as much "work" as possible. Too much doc can confuse readers.

```suggestion
```

> Every Instance with a certificate includes paths to certificates in their configuration files. If you remove one certificate, that change is limited to that one Instance.

Reviewer 2: Hi @mjang, just checking in, what are your thoughts on this part of the doc?

> Every Instance with a certificate includes paths to certificates in their configuration files. If you remove one certificate, that change is limited to that one Instance.

Author: I'd like to avoid references to "Data Plane", as we'll soon incorporate [NGF data planes](https://docs.nginx.com/nginx-gateway-fabric/overview/gateway-architecture/#the-nginx-gateway-fabric-pod) in N1C. 

In any case, I think this is superseded by https://github.com/nginx/documentation/pull/291/files#diff-2445ce44d5814d8ceb795a09fa3672417a9d86c176326cf9ecbc3036b8fc5702R170-R178

If that certificate is already managed as part of a Config Sync Group, the changes you make affect all instances in that group.

## Delete a deployed certificate

You can remove a deployed certificate from an independent instance or from a Config Sync Group. This action also deletes the certificate files or certificate-key pairs from the data plane Instance(s).
<!-- possible common content -->
To do so, navigate to the **Manage** > **Instances** or **Config Sync Groups** menu. Then:

- Select the instance or Config Sync Group where you want to make the change.
- Select the **Configuration** tab.
- Select **Edit Configuration**.
- Find the file that you want to change.
- Select the three-dot "ellipsis" (...), to open a menu. Your options are to
  - **Move/Rename**
  - **Copy**
  - **Delete from Configuration** 
<!-- end possible common content -->

In this case, you can delete a deployed certificate from the configuration.

> Every Config Sync Group also includes paths to managed certificates in its configuration files. If you remove a managed certificate to a Config Sync Group, that change affects all instances which belong to that Config Sync Group. 

Reviewer 2: ```suggestion
Every instance or Config Sync Group might include paths to managed certificates in its configuration files. If you remove a managed certificate to a Config Sync Group, that change affects all instances which belong to that Config Sync Group. 
```
